solutions/src/main/java/ru/sberbank/school/task07/ReverseIteratorImpl.java

```diff
@@ -0,0 +1,65 @@
+package ru.sberbank.school.task07;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+
+/**
+ * 26.05.2019
+ * Реализация простого обратного итератора для обхода списка
+ *
+ * @author Gregory Melnikov
+ */
+
+public class ReverseIteratorImpl<E> implements ReverseOrderIterator<E> {
+
+    private int cursor;
+
+    private final List<E> list;
+
+    public ReverseIteratorImpl(List<E> list) {
+        this.list = list;
+        cursor = list.size();
+    }
+
+    @Override
+    public boolean hasNext() {
+        return cursor != 0;
+    }
+
+    @Override
+    public E next() {
+        if (cursor < 1) {
+            throw new NoSuchElementException();
+        }
+        return list.get(--cursor);
+    }
+}
+
+
+class ReverseOrderIteratorImplSimple<E> implements ReverseOrderIterator<E> {
```

Comment:
Top-level class ReverseOrderIteratorImplSimple has to reside in its own source file.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a102017045>*